src/TaskLineRenderer.ts

```diff
@@ -230,7 +256,7 @@ async function taskToHtml(
 
     // Now build classes for the hidden task components without rendering them
     for (const component of taskLayout.hiddenTaskLayoutComponents) {
-        const hiddenComponentDataAttribute = getComponentDataAttribute(component, task);
+        const hiddenComponentDataAttribute = FieldLayouts[component].getDataAttribute(component, task);
```

Comment:
This and several similar lines is a bit of a smell to me - the need to supply the `component` value twice.

It's a potential source of error, and is what is preventing me from just going ahead and merging this PR.

Reply:
Same thing here. This is refactored already but I propose to submit in a separate PR.

Reply:
> This and several similar lines is a bit of a smell to me - the need to supply the `component` value twice.
> 
> It's a potential source of error, and is what is preventing me from just going ahead and merging this PR.

This is going to be addressed in a later PR - resolving